### DISCUSSION TITLE
Criei uma biblioteca JS para puxar vídeos do YouTube

### DISCUSSION THREAD

**Discussion by ojvribeiro** [frontendbr/forum#2284] [Mostre seu trabalho]
A [YotooJS](https://github.com/ojvribeiro/yotoojs) é pequena e usa a YouTube API v3 para conseguir os dados.

**Comment by felquis** [comment]
Adorei, por que você não mantem a estrutura do json de resposta igual ao original que já vem do youtube?

Achei maneiro que você coloca um embed iframe já, parece que é bem útil para a aplicação que você está resolvendo.

Eu tenho uma página que incorpora vídeos do youtube em tbbc.app/interruptless e sonho em colocar meta tags de twitter cards e aquelas open graph sabe? E seu código é exatamente o que eu preciso fazer, junto com um conjunto de outras decisões de cache.

Mas enfim, muito massa! 🍝

**Reply by ojvribeiro** [reply]
Fala, @felquis

A decisão de mudar a estrutura foi pra simplificar ao máximo o consumo dos dados. A ideia é tornar o mapa simples possível o consumo da API do YouTube via JavaScript. Eu, particularmente, achei a response original bem complicadinha rs. Mas estou aberto a sugestões sempre.

Ah, além do embed, adicionei o campo de shortUrl também. 😁

Enfim, fico muito feliz que gostou do projeto. Agradeço de verdade pelo comentário. Abri o projeto pra comunidade na esperança de torná-lo melhor.
